hivesim-rs/src/testapi.rs

```diff
@@ -360,10 +365,8 @@ pub struct NClientTestSpec {
     pub always_run: bool,
     // The Run function is invoked when the test executes.
     pub run: AsyncNClientsTestFunc,
-    // length of private key array should match clients array
-    // this attribute is optional and is used for tests which need node_ids relative to the
-    // content within the test
-    pub private_keys: Option<Vec<Option<String>>>,
+    // a hashmap of Hive Environment Variables
+    pub environment: Option<Vec<Option<HashMap<String, String>>>>,
```

Comment:
This type has a few layers that seem worth explaining. This comment line seemed a bit helpful (although not really strong enough), seems like it should stay in (or something like it): `// length of private key array should match clients array`

Could be something like:
```suggestion
    /// For each client, there is a distinct map of Hive Environment Variable names to values.
    /// The environments must be in the same order as the `clients`
    pub environments: Option<Vec<Option<HashMap<String, String>>>>,
```

Also, each client gets its own environment, so this Vec with an environment for each client combined into one seems to make sense to be named with the plural `environments`.